### ISSUE TITLE
[ Dev - Feedback] Any cool blueprint to share? Faction Help

### ISSUE THREAD

**Issue by tsunamayo** [tsunamayo/Skywanderers#1569]
Okay I would like to know if some of you had any wip blueprint to share!
I will need to get some cool factions into the game!
I made a doc for those interested:
https://docs.google.com/document/d/e/2PACX-1vT_lv2Ru060nvEa0ggMOLI-v6H1i9snbAt-1lJFe9a2UugRO9C3LaHXzRrBMDl4_chXgpInBaw4iOCc/pub
If you are interested please reply to the thread!
Weapon update is coming soon (hopefully tomorrow)
Cheers and thanks for your help!

**Comment by Aiwii** [comment]
I'm not a Kickstarter backer but I have participated in a backing wave and have access to the game, I would love to build ships for the game and I am totally interested.

Here are some ships i've done (although not complete)
![Skywanderers 2019-10-13 12_19_06 PM](https://user-images.githubusercontent.com/53021785/73023733-63d81000-3de9-11ea-86b6-0f2a1d83e82a.png)
![Skywanderers 2020-01-20 6_04_49 PM](https://user-images.githubusercontent.com/53021785/73023773-705c6880-3de9-11ea-8e55-8d51fc47505e.png)
![Skywanderers 2020-01-18 3_04_14 PM](https://user-images.githubusercontent.com/53021785/73023787-79e5d080-3de9-11ea-99e1-b34af85839ce.png)

and my current masterpiece which you can download (http://skywa.re/blueprint/ok80-wip)
![Layer 0](https://user-images.githubusercontent.com/53021785/73023904-aef22300-3de9-11ea-94ec-7e793858945c.png)

**Comment by Dwarf-LordPangolin** [comment]
First, I'm **_hugely_** excited that we can start submitting stuff! Also, the vast majority of the design rules & tips in that document are awesome; those are all really good design principles. And I'm very happy that your main emphasis is balance first; that will make development go a lot smoother than it otherwise might. (One question: by "faction," are you referring to the big NPC factions, or the backer-designed "corporations?")

There is one comment I need to make, though. Sooo ... I _had_ a bunch of work-in-progress ships ready to send to you when you were ready for them:
![Ships](https://user-images.githubusercontent.com/33041786/73026627-68a3c080-3df7-11ea-8a2c-840cf02509ad.jpg)

However, I'm going to have to rebuild all of them from scratch, because this design requirement presents a bit of a problem:

> We need to have 3m (or more) high corridor and room. 2m feels way too cramped and claustrophobic. 2.5m is out as I will remove the slab.

The reason this is a problem is that all of these use a 2m ceiling. Now, we all agreed to make ships to your specifications, so whatever you ask for, you will get. I'll have to adjust my building style, but that's not a problem. You want 3m minimum ceilings? You will **get** 3m minimum ceilings! :+1: 

The reason this is a problem is because a 2m ceiling is fairly commonly used in voxel building games. A lot of people build hallways that are 2m tall (because the characters' collision model fits in there), and 2m wide (so 2 characters can pass each other). It's a fairly common height for ceilings in 1m voxel games because a lot of people don't like to build taller interiors, because doing so requires a lot more resources. Some people build fancier interiors, but a lot build interiors that meet the minimum volume needed for efficient movement, and nothing else.

If the player characters can fit through a 2m tall space, but the AI doesn't work in a 2m tall space, then once the game is released, players will build compact, efficient ships to match the player character's minimum size requirements, but will then be confused and frustrated when the NPCs can't navigate the same spaces the players can.

So what I'm trying to say is that, if you want 3m tall interiors because you want the ships in the game to be really good-looking and not feel cramped, that's not a problem; but if it's because the AI won't be able to navigate the same space that the players can, that could be an issue down the line. I can definitely see players complaining about that.

- - - - - - - - - -

Since your Google document talks a lot about aesthetic principles of design, I also wanted to mention that there is also an aesthetic element to this. A 50m ship built with a 3m ceiling will look a lot less sleek than a 50m ship with a 2m ceiling. As an example, this is a 50m ship with a 2m ceiling:
![20200115154741_1](https://user-images.githubusercontent.com/33041786/73029605-79573500-3dfd-11ea-867a-e98f7318ae2d.jpg)

It will not look as sleek with a 3m ceiling -- unless I increase the length, but that would make it longer than 50m, and thus too long for its class. So if you want the game's ships to be comparatively tall, that's your design choice, but this will make it harder to make sleeker ships, because of the increase to ships' minimum heights. The minimum height:length ratio of the ships will be different, so of course the overall aesthetics of the ships will be different.

Like I said, you're the boss. I will happily design ships to meet whatever requirements you set; but I feel you should know that increasing the height of the interiors while keeping the length of the exteriors the same is going to make the ships look really ... well, chonky. ;)

**Comment by tsunamayo** [comment]
Okay great thanks for your answer guys and thanks for the bp @Aiwii, very original submarine theme ) ! 
@Dwarf-LordPangolin could you send me the blueprint, I want to check again that 2m ceiling thing. 
I also want to check other details. Your faction looks solid!
It definitely felt too low last time I checked, again for small maintenance corridor it might be a good things as you want them to feel cramped, but not for living / working room.
For the AI there should not be an issue with 2m/3m, but definitely more with brick heavy interior, as collision and pathfinding will likely be voxel based.
Cheers!

**Comment by ZachZent** [comment]
Here is an idea because 3m tall ceilings feels extremely high. It is true that 2m high feels cramped with the 2m tall characters. So why not lower the character models and player camera to 1.75m? That way 2m tall ceilings are no longer awkwardly cramped and 3m won't be awkwardly tall?
![2m](https://user-images.githubusercontent.com/36980446/73038859-3ef79300-3e11-11ea-9c0d-0ca5eeb56422.png)
2m tall
![1 75](https://user-images.githubusercontent.com/36980446/73038874-47e86480-3e11-11ea-8463-e6ed406ebbe8.png)
"1.75"m tall (actual pic is 2.25 high wall/ceiling while character is 2m, but the difference is still the same)

I personally prefer a more cramped interior, utilitarian, efficient. Especially for hallways. If it is only a matter of the feeling of cramped, then lowering the model/camera height would allow for some more cramped areas and slimmer ships. See Dwarfs concern. A maintenance hatch 3m tall would be a little silly. It would also let players who are building for themselves and not for an official ship not get that feeling of claustrophobia solving that problem ahead of time.

**Comment by ExodistSKY1** [comment]
I would like to help out where I can with this! 

I have been working on a Militaristic Faction based on my experience in that occupation.

Here are some of my WIP screenshots:

Just need to finish the turrets and the interior. 525m Length
![20200121225713_1](https://user-images.githubusercontent.com/45324484/73043654-be925b80-3e2b-11ea-9db2-d68a22ca1765.jpg)

![20200108220147_1](https://user-images.githubusercontent.com/45324484/73043827-87707a00-3e2c-11ea-8b1b-38d8fc765011.jpg)
![20191113000911_1](https://user-images.githubusercontent.com/45324484/73043863-acfd8380-3e2c-11ea-9aad-13ab1d5cfa03.jpg)

![20191112230250_1](https://user-images.githubusercontent.com/45324484/73043836-95be9600-3e2c-11ea-8a07-a1fa362ab7a9.jpg)

![20200121221508_1](https://user-images.githubusercontent.com/45324484/73043875-b4249180-3e2c-11ea-8080-7be4b24e012b.jpg)

**Comment by Garrett-C** [comment]
In regards to the hallway thing. I kind of feel like cramped interiors isn’t always a bad thing. Sometimes it’s a look you want. For example the interior of a modern submarine is hardly luxurious. I think 3m in most cases works but a hard and fast no 2m corridors is a bit meh. Having a cramped engineering corridor or something sounds pretty good to me.

**Comment by Garrett-C** [comment]
> Here is an idea because 3m tall ceilings feels extremely high. It is true that 2m high feels cramped with the 2m tall characters. So why not lower the character models and player camera to 1.75m? That way 2m tall ceilings are no longer awkwardly cramped and 3m won't be awkwardly tall?

Interesting suggestion Zach. It might be an idea to have different races have different heights. On top of that, a faction like the scrappy petcha could be shorter and use cramped interiors while a faction like the imperitus could be the 2m ones with tight but livable interiors.

**Comment by ZachZent** [comment]
> Interesting suggestion Zach. It might be an idea to have different races have different heights. On top of that, a faction like the scrappy petcha could be shorter and use cramped interiors while a faction like the imperitus could be the 2m ones with tight but livable interiors.

Just for fun, I looked up average heights. Well American average heights, but still a decent representation. The 2m character model would be at the far end of the bell curve where 1.75m is actually in a very average position. Can even go up or down for a noticeable height difference in different species and still be below the 2.0m mark. Love to see some tiny aliens lower on the bell curve.
![unknown](https://user-images.githubusercontent.com/36980446/73045503-69564a00-3e2b-11ea-8f1f-f3ec7a4a2552.png)

**Comment by Garrett-C** [comment]
>The 2m character model would be at the far end of the bell curve where 1.75m is actually in a very average position.

That's a fair point. However with this being scifi I think it is fair to streach that a little bit. Firstly because humans have been getting taller for decades so one could argue it's just an extension of that and secondly there is always bioengineering. A distinctly militaristic and authoritarian looking race like the imperitus would probably be all over genetic modification to look tougher. But other races like the plant guys could be taller. You could even have the mushroom people shown off in concept art being really short. Basically yeah it would be neat to see a range of heights for player models.

**Comment by Dwarf-LordPangolin** [comment]
Got a blueprint ready to send your way!
![20200124221957_1](https://user-images.githubusercontent.com/33041786/73116187-af78df80-3ef7-11ea-9c68-fe51aebd7185.jpg)

This version has had the interior removed and resized for 3m high ceilings as requested. It's all function; no fancy interior, and all the brick detailing on the exterior has been removed. Obviously, a release-ready version would look nicer, but this is purely for testing performance & balance.

[Starwave Corvette - 1st Testing Version.zip](https://github.com/tsunamayo/Skywanderers/files/4111614/Starwave.Corvette.-.1st.Testing.Version.zip)

**Comment by tsunamayo** [comment]
Hi, thanks for your contribution all! There is some nice ship in here, I am excited to try everything out!

So regarding the ceiling height, I agree 3m is too much. But 2m too low either, except when being cramped is the goal (like maintenance corridor, cramped quarter, ect). And between 3m and 2m I just prefer 3! 
On your last two picture I prefer the last one...
But for sure doors needs to be 3m. Like in real house, the door dont go up to the top.
Also I like to put some gribling to the top, to make it 2.5m which could be a great height. Put down some pipes, grating or ramp block and then you get something visually interesting.
Do you have the 2m version of the Starwave so I can compare? As you did both 2m and 3m I thing this is the perfect opportunity to arbitrate this once and for all.

Cheers - and thanks for the constructive discussion!

**Comment by Dwarf-LordPangolin** [comment]
So, when you said "3m or more" for the ceilings, I thought that meant 3m _unobstructed_ height -- that we need to have 3m of space with nothing in it, not even greebling. But since you clarified [here](https://github.com/tsunamayo/Skywanderers-Pathfinder-Group-Build/issues/1) that we can put greebles inside that 3m space, I don't have any more concerns about the 3m height. 3m w/detailing looks better than both 2m, and 3m w/o detailing. Now that I understand what you meant, I think we're on the same page about ceilings.

And I can send you the old Starwave if you want, but that one was too cramped even for me! It had a very pretty interior, but it was meant for looks, so the hallway was much too narrow.
![20200115162337_1](https://user-images.githubusercontent.com/33041786/73387123-abe9ad80-4295-11ea-92cf-46cf59e01ea4.jpg)

**Comment by ExodistSKY1** [comment]
I agree with Dwarf, now knowing we can add detail to the 3m height makes the 3m perfect for the best of both worlds

**Comment by tsunamayo** [comment]
Yes sorry, the 3m was not really for AI, it was for the cramped feeling. Your interior style is really clean and good! Just too cramped ;)
The important thing for AI will be: using block interior, not too much decoration brick inside as collision model will likely be simpler. I think I would add debugging tool to see where AI can go or not.

This is quick dirty test I did back in the day, with current tech I would be unable to use the unity solution - I will have to roll my own one...

https://www.youtube.com/watch?v=R1PL1ylKxUc
https://www.youtube.com/watch?v=0PEXqSYl7dc
https://www.youtube.com/watch?v=LFHCmEE_Pz8

I am gonna release the crew doc I made to get some feedback I think.
Cheers

**Comment by Majikmonster** [comment]
I just want to confirm: Are blueprint submissions only for Kickstarter backers, or is this open to all backers? The document mentions Pathfinders, which is why I'm asking.

I would also like to know why the half slab is being cut from the game.

**Comment by JRL101** [comment]
Heres one i've been playing with, of corse its not perfect, but i've been playing with it all day, messing with different doors.  Needs landing gear, but going to make it simple to minimise moving subgrids. but still having a functional ship for a player

![image](https://user-images.githubusercontent.com/33255536/82644759-6ab24a80-9c44-11ea-93e3-7e076023b148.png)
![image](https://user-images.githubusercontent.com/33255536/82644779-7736a300-9c44-11ea-80e0-f0381494d28a.png)
![image](https://user-images.githubusercontent.com/33255536/82644896-a0efca00-9c44-11ea-9628-0639b8af989b.png)
![image](https://user-images.githubusercontent.com/33255536/82644926-ae0cb900-9c44-11ea-923b-65f0e77d2b7d.png)
![image](https://user-images.githubusercontent.com/33255536/82644986-c250b600-9c44-11ea-8276-273441b45d36.png)
![image](https://user-images.githubusercontent.com/33255536/82645020-ced50e80-9c44-11ea-8714-f29d6e707884.png)
![image](https://user-images.githubusercontent.com/33255536/82645064-e01e1b00-9c44-11ea-9bb3-2bf9d1532e11.png)
![image](https://user-images.githubusercontent.com/33255536/82645110-f035fa80-9c44-11ea-9851-33280dbcc3e8.png)


[Pirate Runner 01a_JRL101.zip (Broken Engins)](https://github.com/tsunamayo/Starship-EVO/files/4666782/Pirate.Runner.01a_JRL101.zip)
EDIT: working engines
[Pirate Runner 01b_JRL101.zip](https://github.com/tsunamayo/Starship-EVO/files/4666904/Pirate.Runner.01b_JRL101.zip)
